api.go

```diff
@@ -51,21 +51,21 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 	}
 
 	// Initialize the network.
-	err = p.network.init(&(p.config.NetworkConfig))
+	netNsPath, netNsCreated, err := p.network.init(p.config.NetworkConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	// Execute prestart hooks inside netns
-	err = p.network.run(p.config.NetworkConfig.NetNSPath, func() error {
+	err = p.network.run(netNsPath, func() error {
```

Comment:
I am not sure to understand all consequences here. How does that solve the fact that we have to be in the netns ? We still need to get this PID in the right netns, so that docker can retrieve the right netns when it is looking for /proc/PID/ns/net, right ? Otherwise I am missing something that I don't understand :)